src/stores/businessBootstrap.ts

```diff
@@ -108,16 +109,29 @@ export const useBcrosBusinessBootstrap = defineStore('bcros/businessBootstrap',
       console.error(`Attempted to load ${identifier} as a bootstrap filing.`)
       return
     }
+
+    const storeIsLoading = new Promise((resolve) => {
+      watch(isStoreLoading, (newValue) => {
+        if (newValue === false) { // check the condition
+          resolve(true)
+        }
+      }, { immediate: true })
+    })
+
+    await storeIsLoading
```

Comment:
No, that is actually a semaphore :D haha.
It waits for the above promise to finish. If above promise is not finished, it will not let the 2nd thread go.

What does this do is, we have calls to `loadBusinessBootstrap` in two places, and why was the issue with not loading stuff in Cypress is because they would step on each other toes, and some stuff would be set, and some would end up unset because of the 2nd call. With this, I make sure that as soon as the call starts, it waits for the 1st call to finish, then checks if it needs to do updates or what not because 2nd call has set either force or is for different business.

Reply:
but there is a caching check so it shouldn't do that?

Reply:
I guess in cypress the 2nd call may trigger too quickly, but I don't think we should expose `isStoreLoading` below in the return (it should be an internal var to the store otherwise someone might add code that updates it and causes issues here) and can you add a comment here pls

Reply:
for sure, I made the store variable store local only.